src/Sulu/Bundle/MarkupBundle/Markup/Link/LinkProviderPool.php

```diff
@@ -47,7 +47,10 @@ public function getConfiguration()
     {
         $configuration = [];
         foreach ($this->providers as $name => $provider) {
-            $configuration[$name] = $provider->getConfiguration();
+            /** @var ?LinkConfiguration|LinkConfigurationBuilder $providerConfiguration */
+            $providerConfiguration = $provider->getConfiguration();
+
+            $configuration[$name] = $providerConfiguration instanceof LinkConfigurationBuilder ? $providerConfiguration->getLinkConfiguration() : $providerConfiguration;
```

Comment:
Add a deprecation warning when the configuration is given instead of the Builder